app/authentication/auth.py

```diff
@@ -154,6 +154,12 @@ def _auth_with_api_key(api_key, service):
             service_id=service.id,
             api_key_id=api_key.id,
         )
+    
+    # TODO: uncomment this when the grace period for the token prefix is over
```

Comment:
Adding the validation logic now even though we won't use it as the mechanism is fresh in my mind.  If you would prefer this not to be here let me know.